observability/metrics/src/test/lib/scala/com/daml/metrics/api/testing/InMemoryMetricsFactory.scala

```diff
@@ -21,48 +21,106 @@ import com.daml.metrics.api.testing.InMemoryMetricsFactory.{
   InMemoryHistogram,
   InMemoryMeter,
   InMemoryTimer,
+  MetricsState,
 }
 import com.daml.metrics.api.{MetricHandle, MetricName, MetricsContext}
 import com.daml.scalautil.Statement.discard
 
 import scala.collection.concurrent.{TrieMap, Map => ConcurrentMap}
+import scala.collection.mutable
 
 class InMemoryMetricsFactory extends LabeledMetricsFactory {
 
-  val asyncGauges: ConcurrentMap[(MetricName, MetricsContext), () => Any] =
-    TrieMap[(MetricName, MetricsContext), () => Any]()
+  val metrics: MetricsState =
+    MetricsState(
+      timers = mutable.Map.empty,
+      gauges = mutable.Map.empty,
+      meters = mutable.Map.empty,
+      counters = mutable.Map.empty,
+      histograms = mutable.Map.empty,
```

Comment:
I don't think we have any guaranty that multiple metrics won't be added at the same time. So those maps likely should be `TrieMap` too.

Reply:
Good point, migrated everything to concurrent maps